packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -156,22 +156,27 @@ abstract class ButtonStyleButton extends StatefulWidget {
   /// {@macro flutter.material.ButtonStyleButton.iconAlignment}
   final IconAlignment iconAlignment;
 
-  /// Returns a non-null [ButtonStyle] that's based primarily on the [Theme]'s
-  /// [ThemeData.textTheme] and [ThemeData.colorScheme].
+  /// Returns a [ButtonStyle] that's based primarily on the [Theme]'s
+  /// [ThemeData.textTheme] and [ThemeData.colorScheme], but has most values
+  /// filled out (non-null).
   ///
-  /// The returned style can be overridden by the [style] parameter and
-  /// by the style returned by [themeStyleOf]. For example the default
-  /// style of the [TextButton] subclass can be overridden with its
-  /// [TextButton.style] constructor parameter, or with a
-  /// [TextButtonTheme].
+  /// The returned style can be overridden by the [style] parameter and by the
+  /// style returned by [themeStyleOf]. For example the default style of the
+  /// [TextButton] subclass can be overridden with its [TextButton.style]
+  /// constructor parameter, or with a [TextButtonTheme].
   ///
-  /// Concrete button subclasses should return a ButtonStyle that
-  /// has no null properties, and where all of the [MaterialStateProperty]
-  /// properties resolve to non-null values.
+  /// Concrete button subclasses should return a [ButtonStyle] with as many
+  /// non-null properties as possible, where all of the non-null
+  /// [MaterialStateProperty] properties resolve to non-null values. Properties
+  /// like [ButtonStyle.fixedSize] and [ButtonStyle.side],
+  /// [ButtonStyle.backgroundBuilder] and [ButtonStyle.foregroundBuilder] may be
+  /// null, since they would override other values in the same [ButtonStyle] if
```

Comment:
I don't think they all override other properties in the ButtonStyle although fixedSize does. Unfortunately a longer explanation might be needed.

Reply:
Maybe I should just enumerate the properties that are allowed to be null and say why for each of them?

Reply:
I think that would help, yes.